array.go

```diff
@@ -1022,3 +1023,123 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// IsOrdered succeeds if array is ordered based on optional `comparator` function.
+// For default, it will use built-in comparator function for each data type.
+// It requires all elements in the array to have same data type
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{100, 101, 102})
+//	array.IsOrdered(func(x, y *Value) bool {
+//		valX, _ := x.Raw().(int)
+//		valY, _ := y.Raw().(int)
+//		return valX >= valY
+//	})
+func (a *Array) IsOrdered(less ...func(x, y *Value) bool) *Array {
+	a.chain.enter("IsOrdered()")
+	defer a.chain.leave()
+
+	if a.chain.failed() {
+		return a
+	}
+
+	if len(less) > 1 {
+		a.chain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected multiple less arguments"),
+			},
+		})
+		return a
+	}
+
+	var fn func(x, y *Value) bool
+	if len(less) == 1 {
+		fn = less[0]
+	} else {
+		// use default comparator based on data type
+		cArr, ok := canonArray(a.chain, a.value)
+		if !ok {
+			return a
+		}
+
+		var currType, prevType string
+		for _, v := range cArr {
+			switch t := v.(type) {
+			case bool:
+				currType = fmt.Sprintf("%T", t)
+				fn = func(x, y *Value) bool {
+					// TODO: implement built-in comparator function for Boolean
+					return (!x.Boolean().value && y.Boolean().value) ||
+						(x.Boolean().value == y.Boolean().value)
+				}
+			case float64:
+				currType = fmt.Sprintf("%T", t)
+				fn = func(x, y *Value) bool {
+					x.Number().Le(y)
+					return !x.chain.failed()
```

Comment:
This approach is not a very good idea. Instead of invoking x.Number / x.String etc, and then calling Le(), it would be better to compare values directly (cast them to float and compare floats).

Why:

* if x is not a number, x.Number will cause test failure; instead of failure in Array.IsOredered, we'll get failure in Value.Number, which will be confusing to user

* even if there is no failure, x.Number will report succeeded assertion to log; it means that the user calls Array.IsOrdered and it causes additional clutter in logs with assertions which the user did not issue directly; this may be confusing too